tests/e2e/tests/content-manager/history.spec.ts

```diff
@@ -7,40 +7,71 @@ import { waitForRestart } from '../../utils/restart';
 
 const edition = process.env.STRAPI_DISABLE_EE === 'true' ? 'CE' : 'EE';
 
+const AUTHOR_CREATE_URL =
+  /\/admin\/content-manager\/collection-types\/api::author.author\/create(\?.*)?/;
+const AUTHOR_EDIT_URL =
+  /\/admin\/content-manager\/collection-types\/api::author.author\/(?!create)[^/]/;
+const ARTICLE_CREATE_URL =
+  /\/admin\/content-manager\/collection-types\/api::article.article\/create(\?.*)?/;
+const ARTICLE_LIST_URL = /\/admin\/content-manager\/collection-types\/api::article.article(\?.*)?/;
+const ARTICLE_EDIT_URL =
+  /\/admin\/content-manager\/collection-types\/api::article.article\/[^/]+(\?.*)?/;
+const ARTICLE_HISTORY_URL =
+  /\/admin\/content-manager\/collection-types\/api::article.article\/[^/]+\/history(\?.*)?/;
+const HOMEPAGE_EDIT_URL = /\/admin\/content-manager\/single-types\/api::homepage.homepage(\?.*)?/;
+const HOMEPAGE_HISTORY_URL =
+  /\/admin\/content-manager\/single-types\/api::homepage.homepage\/history(\?.*)?/;
+
 const goToHistoryPage = async (page: Page) => {
-  const moreActionsButton = await page.getByRole('button', { name: /more actions/i });
+  const moreActionsButton = page.getByRole('button', { name: /more actions/i });
   await expect(moreActionsButton).toBeEnabled();
   await moreActionsButton.click();
-  const historyButton = await page.getByRole('menuitem', { name: /content history/i });
-  await historyButton.click();
+  const historyButton = page.getByRole('menuitem', { name: /content history/i });
+
+  // TODO find out why the history button sometimes doesn't appear. Reloading shouldn't be necessary
+  if (await historyButton.isVisible()) {
+    await historyButton.click();
+  } else {
+    await page.reload();
+    await goToHistoryPage(page);
+  }
+};
+
+/**
+ * The success notification lets us make sure an action is completed before moving on.
+ * We also close it to make sure only one appears at a time, otherwise the test fails.
+ */
+const closeSuccessNotification = async (page: Page) => {
+  const successNotification = page.getByRole('alert', { name: /success/i });
+  if (await successNotification.isVisible()) {
+    await successNotification.getByRole('button', { name: /close/i }).click();
+  }
 };
 
 describeOnCondition(edition === 'EE')('History', () => {
-  test.describe('Collection Type', () => {
-    test.beforeEach(async ({ page }) => {
-      await resetDatabaseAndImportDataFromPath('with-admin.tar');
-      await resetFiles();
-      await page.goto('/admin');
-      await login({ page });
-    });
+  test.beforeEach(async ({ page }) => {
+    await resetDatabaseAndImportDataFromPath('with-admin.tar', (cts) => cts, { coreStore: false });
```

Comment:
I'm sure there is a reason but I'm struggling to understand why this config was needed and set to false here.

Reply:
Currently on v5/main the core_store isn't reset by the function between tests.

core_store is used for "configure the view changes", and for us it meant that when we have a test the deletes a field from the layout, the test that runs after that doesn't have the input.

I feel like the core_store should always be reset, but the way it was configured looked intentional, so I chose to modify it only for us via a param to avoid breaking other tests, at least for now